### PR TITLE
Speedup with Cython and Turtle parsing.

### PR DESCRIPTION
**Pull request by rchateauneu**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: rdflib/plugins/parsers/notation3.py [hunk 1]
```diff
@@ -1570,8 +1585,8 @@ def strconst(self, argstr, i, delim):
             if ch == delim1:
                 j = i
                 continue
-            elif ch in ('"', "'") and ch != delim1:
-                ustr = ustr + ch
+            elif ch in "\"'" and ch != delim1:
```
Comment (FlorianLudwig) [inline rdflib/plugins/parsers/notation3.py:1588]: To me the original looks more readable and - at least in cpython - it is actually faster. Didn't test cython though.

**Comment by FlorianLudwig** [comment]
Overall this looks like a good MR to me. A lot of cleanup and cython :+1: 

I am curious why travis fails on powerpc - I cannot see any errors in the logs. Is it just me being blind?

**Comment by FlorianLudwig** [comment]
Maybe it might be a good idea to split this up into two merge requests, one for the cleanup of the python code and one for setting up cython.

**Comment by rchateauneu** [comment]
> 
> 
> It might be a good idea to split this up into two merge requests, one for the cleanup of the python code and one for setting up cython.

Yes, why not. I am discovering rdflib code, so sorry for these errors.

Please do not hesitate to separately take any change of think is worth.

**Comment by rchateauneu** [comment]
> 
> 
> Overall this looks like a good MR to me. A lot of cleanup and cython +1
> 
> I am curious why travis fails on powerpc - I cannot see any errors in the logs. Is it just me being blind?

And, I restored setup.py to what it was, and I get the same error message on ppc only:
/plugins/serializers/nt.py:28: UserWarning: NTSerializer does not use custom encoding.

  warnings.warn("NTSerializer does not use custom encoding.")

/home/travis/build/rchateauneu/rdflib/rdflib/plugins/serializers/nquads.py:24: UserWarning: NQuadsSerializer does not use custom encoding.

  warnings.warn("NQuadsSerializer does not use custom encoding.")

.....

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

File: rdflib/plugins/parsers/notation3.py (hunk 1 above)
Comment (rchateauneu) [reply rdflib/plugins/parsers/notation3.py:1588]: Indeed the effect of replacing "x = x + y" by "x += y" is very small. However, the other changes have a significant effect on my platform.

**Comment by FlorianLudwig** [comment]
@ddeka2910 can you help here?

**Comment by rchateauneu** [comment]
Thanks Ludwig. I have recreated a new clone from scratch, but the problem still happens.

**Comment by white-gecko** [comment]
I have converted this one to a draft as the turtle parsing part is covered in #1266 and this one should be changes to only cover the cython improvements.

**Comment by white-gecko** [comment]
That would be great

**Pull request closed without merging**